docs/querying/sql-data-types.md

```diff
@@ -34,12 +34,23 @@ Druid associates each column with a specific data type. This topic describes sup
 
 Druid natively supports the following basic column types:
 
-* LONG: 64-bit signed int
-* FLOAT: 32-bit float
-* DOUBLE: 64-bit float
-* STRING: UTF-8 encoded strings and string arrays
-* COMPLEX: non-standard data types, such as nested JSON, hyperUnique and approxHistogram, and DataSketches
-* ARRAY: arrays composed of any of these types
+* `LONG`: 64-bit signed int
+* `FLOAT`: 32-bit float
+* `DOUBLE`: 64-bit float
+* `STRING`: UTF-8 encoded strings and string arrays
+* `ARRAY`: arrays composed of any of these types
+
+## Complex types
+
+Druid natively supports the following complex types:
+* `COMPLEX<JSON>`: stores a copy of structured data in JSON format and specialized internal columns and indexes for nested basic types. Click here to learn more about [`COMPLEX<JSON>`](nested-columns.md)
+* `cardinality`: Data structure to compute the cardinality of Apache Druid dimensions using the HyperLogLog algorithm. Click here to learn more about [`cardinality`](hll-old.md#cardinality-aggregator)
+* `hyperUnique`: Data structure of aggregated values to estimate count distinct using a variant of the HyperLogLog approximation algorithm. Consider using HLL sketches for better accuracy in many cases. Click here to learn more about [`hyperUnique`](hll-old.md#hyperunique-aggregator)
```

Comment:
if we are going to mention json as `COMPLEX<json>`, then this should probably be `COMPLEX<hyperUnique>`?

Also its strange to tell someone to use something and not also include a description of that type... I'm not sure how we should handle extension `COMPLEX` types here, since there are a lot of them and I don't know that their docs do a very good job of articulating what type they store is if they are ingested into a rollup table

Reply:
btw, here is a list i put together earlier when discussing how to document `decode_base64_complex`

```
built-in
------------------
hyperUnique
serializablePairLongString

druid-bloom-filter
------------------
bloom

druid-datasketches
-----------------
arrayOfDoublesSketch
HLLSketch
KllDoublesSketch
KllFloatsSketch
quantilesDoublesSketch
thetaSketch

druid-histogram
-----------------
approximateHistogram
fixedBucketsHistogram

druid-stats
-----------------
variance

druid-compressed-big-decimal
------------------
compressedBigDecimal

druid-momentsketch
------------------
momentSketch

druid-tdigestsketch
------------------
tDigestSketch
```

though it might be missing some types, for example i notice spectator histogram is missing from this list

Reply:
I think keeping it consistent with the COMPLEX wrapper is a smart idea.

I think linking to their docs makes the most sense. Even if their doc's are/are not that good, its would still be the best place to learn about the Complex type?

Reply:
the extension docs primarily describe how to use the extension functions. It would make sense to me to list out the types here and link each of them to their dedicated docs to learn more about how to create and use these types